test/e2e-tests.js

```diff
@@ -218,84 +204,103 @@ describe('matrix-public-archive', () => {
       // TODO: Set avatar of room
 
       // Test image
-      const mxcUri = await client.uploadContentFromUrl(
-        'https://en.wikipedia.org/wiki/Friction#/media/File:Friction_between_surfaces.jpg'
+      // via https://en.wikipedia.org/wiki/Friction#/media/File:Friction_between_surfaces.jpg (CaoHao)
+      const imageBuffer = await readFile(
+        path.resolve(__dirname, './fixtures/friction_between_surfaces.jpg')
       );
-      const imageEventId = await client.sendMessage(roomId, {
-        body: 'Friction_between_surfaces.jpeg',
-        info: {
-          size: 396644,
-          mimetype: 'image/jpeg',
-          thumbnail_info: {
-            w: 800,
-            h: 390,
+      const imageFileName = 'friction_between_surfaces.jpg';
+      const mxcUri = await uploadContent({
+        client,
+        roomId,
+        data: imageBuffer,
+        fileName: imageFileName,
+      });
+      const imageEventId = await sendMessageOnArchiveDate({
+        client,
+        roomId,
+        content: {
+          body: imageFileName,
+          info: {
+            size: 17471,
             mimetype: 'image/jpeg',
-            size: 126496,
+            w: 640,
+            h: 312,
+            'xyz.amorgan.blurhash': 'LkR3G|IU?w%NbxbIemae_NxuD$M{',
           },
-          w: 1894,
-          h: 925,
-          'xyz.amorgan.blurhash': 'LkR3G|IU?w%NbwbIemae_NxuD$M{',
-          // TODO: How to get a proper thumnail URL that will load?
-          thumbnail_url: mxcUri,
+          msgtype: 'm.image',
+          // TODO: How can we get the thumbnail to load? The full lightbox version works.
```

Comment:
We need to enable `dynamic_thumbnails` in `homserver.yaml`. This was a bit non-obvious to know because Synapse was just throwing out 404's before. I've created an issue to make the error more obvious and friendly, https://github.com/matrix-org/synapse/issues/13016